ProxyStarcraft/ZergUnitExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections.Generic;
+
+namespace ProxyStarcraft
+{
+    public static class ZergUnitExtensions
+    {
+        public static void SpawnLarva(this ZergUnit queen, ZergBuilding hatchery, List<Command> commands)
+        {
+            if (queen == null || hatchery == null) // TODO: I had a queen die, I guess mid-update, causing a Null Ref.
```

Comment:
If you got a null reference, it wasn't because something died, because you only take actions between updates. There's no scenario under which you would have a null reference in your code because of what happened between getting one update and getting the next.

Reply:
This *may* have been the Steps being out of order. (Terran bot issues commands to server, steps, and then zerg bot (using the previous step's game state, issues a command to the now-dead queen, which the server can't reconcile-- it would be nice if the server ignored commands to recently-dead units!)

I'll remove this for now, and if we run into it later, deal with it then.